hudi-flink/src/main/java/org/apache/hudi/streamer/HoodieFlinkStreamer.java

```diff
@@ -93,8 +95,15 @@ public static void main(String[] args) throws Exception {
         ), kafkaProps))
         .name("kafka_source")
         .uid("uid_kafka_source")
-        .map(new RowDataToHoodieFunction<>(rowType, conf), TypeInformation.of(HoodieRecord.class))
-        // Key-by record key, to avoid multiple subtasks write to a partition at the same time
+        .map(new RowDataToHoodieFunction<>(rowType, conf), TypeInformation.of(HoodieRecord.class));
+    if (conf.getBoolean(FlinkOptions.INDEX_BOOTSTRAP_ENABLED)) {
```

Comment:
1. Since this is a very time-consuming operation， we'd better add a 'TODO'  here, indicating that optimization is needed here.
2. Maybe we should remind the user `HoodieDeltaStreamer` is not suitable to initialize on large tables when we have no checkpoint to restore from, Another pr on the docs of the website